File: examples/test-terry-davis.go
/**
 * Terry Davis style Go - unconventional, playful, surprising
 * "God said this should work"
 * Run with: go run test-terry-davis.go
 */

package main

import (
	"fmt"
	"math/rand"
	"time"
)

// God said this should work
const (
	GodMode      = true
	SatanMode    = false
	TempleValue  = 5*5 + 0x666
)

// Cosmic variable names
var (
	GodPointer    []int
	DivineArray   = [5]int{1, 2, 3, 4, 5}
	HolyCVariable = 42
	TempleOSKernel = "TempleOS 5.017"
	SatanBuffer   []int
	HeavenlyLoop  int
)

// WARNING: Satan tries to corrupt this buffer
func SacredFunction(n int) []int {
	// God said this should work
	GodPointer = make([]int, n)
	result := make([]int, n)
	for i := 0; i < n; i++ {
		result[i] = i * i
	}
	return result
}

// Goto-style control flow (Go doesn't have goto, but we can simulate)
func ProcessDataWithGoto() {
	fmt.Println("The chosen path")
	
	// Simulate goto with goto (Go actually has goto!)
	goto step2
	
step1:
	fmt.Println("God's path not taken")
	goto end
	
step2:
	if rand.Float64() > 0.5 {
		goto step1
	}
	fmt.Println("Blessed completion")
	
end:
	return
}

// Recursive function - infinite temple recursion
func RecursiveTemple(depth, maxDepth int) {
	if depth < maxDepth {
		fmt.Printf("Temple level %d\n", depth)
		RecursiveTemple(depth+1, maxDepth)
	} else {
		fmt.Println("We have reached the inner sanctum")
	}
}

// Obfuscated one-liner
func ObfuscatedOneLiner() {
	// Print 0-9 with closure
	for i := 0; i < 10; func() { fmt.Print(i, " "); i++() } {
	}
	fmt.Println()
}

// Direct memory manipulation (using pointers)
func DirectMemoryDemo() {
	memory := make([]int, 10)
	
	// Write to "memory"
	for i := range memory {
		memory[i] = i * 100
	}
	
	// Read from "memory"
	fmt.Print("Memory contents: ")
	for _, val := range memory {
		fmt.Print(val, " ")
	}
	fmt.Println()
}

// Interface-based prototype manipulation
type DivineBehavior interface {
	Sacred(x int) int
	Temple(s string) string
}

type DivineObject struct{}

func (d DivineObject) Sacred(x int) int {
	return x * x
}

func (d DivineObject) Temple(s string) string {
	return fmt.Sprintf("Temple: %s", s)
}

// Main execution
func main() {
	rand.Seed(time.Now().UnixNano())
	
	fmt.Println("Terry Davis style Go demonstration")
	fmt.Println("===================================")
	fmt.Println()
	
	fmt.Println("Operating System:", TempleOSKernel)
	fmt.Println("HolyC Variable:", HolyCVariable)
	
	// God mode activated
	if GodMode {
		fmt.Println("God mode activated")
	}
	
	// This never runs
	if SatanMode {
		fmt.Println("This never runs")
	}
	
	fmt.Println("Temple value:", TempleValue)
	
	// Sacred function
	result := SacredFunction(5)
	fmt.Print("Sacred array: ")
	for _, val := range result {
		fmt.Print(val, " ")
	}
	fmt.Println()
	
	// Process data with goto simulation
	ProcessDataWithGoto()
	
	// Recursive temple
	RecursiveTemple(0, 5)
	
	// Divine array
	divineArray := DivineArray
	divineArray[4] = 6 // Modify last element
	fmt.Print("Divine array: ")
	for _, val := range divineArray {
		fmt.Print(val, " ")
	}
	fmt.Println()
	
	// Direct memory demo
	DirectMemoryDemo()
	
	// Interface demo
	var divineObj DivineObject = DivineObject{}
	fmt.Printf("Sacred(5) = %d\n", divineObj.Sacred(5))
	fmt.Printf("Temple(\"test\") = %s\n", divineObj.Temple("test"))
	
	// Obfuscated one-liner
	ObfuscatedOneLiner()
}

File: examples/test-terry-davis.rs
/**
 * Terry Davis style Rust - unconventional, playful, surprising
 * "God said this should work"
 * Compile with: rustc test-terry-davis.rs
 * Run with: ./test-terry-davis
 */

use std::collections::HashMap;

// God said this should work
macro_rules! GOD_MODE {
    () => {
        if true {
            println!("God mode activated");
        }
    };
}

macro_rules! SATAN_MODE {
    () => {
        while false {
            println!("This never runs");
        }
    };
}

macro_rules! TEMPLE {
    ($x:expr) => {
        $x * $x + 0x666
    };
}

// Cosmic variable names
static mut GOD_POINTER: Option<Vec<i32>> = None;
static DIVINE_ARRAY: [i32; 5] = [1, 2, 3, 4, 5];
static HOLY_C_VARIABLE: i32 = 42;
static TEMPLE_OS_KERNEL: &str = "TempleOS 5.017";
static mut SATAN_BUFFER: Vec<i32> = Vec::new();
static mut HEAVENLY_LOOP: i32 = 0;

// WARNING: Satan tries to corrupt this buffer
fn sacred_function(n: usize) -> Vec<i32> {
    // God said this should work
    unsafe {
        GOD_POINTER = Some(Vec::with_capacity(n));
    }
    
    let mut result = Vec::with_capacity(n);
    for i in 0..n {
        result.push((i * i) as i32);
    }
    result
}

// Goto-style control flow (Rust doesn't have goto, but we can simulate)
fn process_data_with_goto() {
    println!("The chosen path");
    
    // Simulate goto with loop and break
    'step2: loop {
        if rand::random::<f64>() > 0.5 {
            println!("God's path not taken");
            break 'step2;
        }
        println!("Blessed completion");
        break;
    }
}

// Recursive function - infinite temple recursion
fn recursive_temple(depth: i32, max_depth: i32) {
    if depth < max_depth {
        println!("Temple level {}", depth);
        recursive_temple(depth + 1, max_depth);
    } else {
        println!("We have reached the inner sanctum");
    }
}

// Obfuscated one-liner
fn obfuscated_one_liner() {
    // Print 0-9 with iterator
    println!("{}", (0..10).map(|i| i.to_string()).collect::<Vec<_>>().join(" "));
}

// Direct memory manipulation (using unsafe)
unsafe fn direct_memory_demo() {
    let mut memory = vec![0i32; 10];
    
    // Write to "memory"
    for i in 0..memory.len() {
        memory[i] = (i * 100) as i32;
    }
    
    // Read from "memory"
    print!("Memory contents: ");
    for val in &memory {
        print!("{} ", val);
    }
    println!();
}

// Trait-based prototype manipulation
trait DivineBehavior {
    fn sacred(&self, x: i32) -> i32;
    fn temple(&self, s: &str) -> String;
}

struct DivineObject;

impl DivineBehavior for DivineObject {
    fn sacred(&self, x: i32) -> i32 {
        x * x
    }
    
    fn temple(&self, s: &str) -> String {
        format!("Temple: {}", s)
    }
}

// Main execution
fn main() {
    println!("Terry Davis style Rust demonstration");
    println!("===================================");
    println!();
    
    println!("Operating System: {}", TEMPLE_OS_KERNEL);
    println!("HolyC Variable: {}", HOLY_C_VARIABLE);
    
    // God mode activated
    GOD_MODE!();
    
    // This never runs
    SATAN_MODE!();
    
    println!("Temple value: {}", TEMPLE!(5));
    
    // Sacred function
    let result = sacred_function(5);
    print!("Sacred array: ");
    for val in &result {
        print!("{} ", val);
    }
    println!();
    
    // Process data with goto simulation
    process_data_with_goto();
    
    // Recursive temple
    recursive_temple(0, 5);
    
    // Divine array
    let mut divine_array = DIVINE_ARRAY.to_vec();
    divine_array.push(6);
    print!("Divine array: ");
    for val in &divine_array {
        print!("{} ", val);
    }
    println!();
    
    // Direct memory demo
    unsafe {
        direct_memory_demo();
    }
    
    // Trait-based demo
    let divine_obj = DivineObject;
    println!("Sacred(5) = {}", divine_obj.sacred(5));
    println!("Temple(\"test\") = {}", divine_obj.temple("test"));
    
    // Obfuscated one-liner
    obfuscated_one_liner();
}